File: src/main/java/de/bhclub/ticker/playlists/Playlist.java
package de.bhclub.ticker.playlists;


import de.bhclub.ticker.gif.Gif;
import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
public class Playlist {

    @Id
    @GeneratedValue
    private long id;

    private String name;

    private boolean shuffle;

    private double waitTime;

    @ManyToMany
    private List<Gif> entries = new ArrayList<>();
}

File: src/main/kotlin/de/bhclub/ticker/playlists/Playlist.kt
package de.bhclub.ticker.playlists

import de.bhclub.ticker.gif.Gif
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.ManyToMany
import java.util.UUID

@Entity
data class Playlist(
    @Id
    val id: UUID = UUID.randomUUID(),
    val name: String,
    var shuffle: Boolean = false,
    val waitTime: Int = 0,

    @ManyToMany
    val entries: MutableList<Gif> = mutableListOf(),
)